core/src/test/java/ai/timefold/solver/core/preview/api/variable/declarative/task_assignment/TAShadowVariableTest.java

```diff
@@ -0,0 +1,72 @@
+package ai.timefold.solver.core.preview.api.variable.declarative.task_assignment;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+
+import ai.timefold.solver.core.api.solver.SolverFactory;
+import ai.timefold.solver.core.config.solver.EnvironmentMode;
+import ai.timefold.solver.core.config.solver.PreviewFeature;
+import ai.timefold.solver.core.config.solver.SolverConfig;
+import ai.timefold.solver.core.config.solver.termination.TerminationConfig;
+import ai.timefold.solver.core.impl.testdata.domain.declarative.task_assignment.TestdataTAConstraintProvider;
+import ai.timefold.solver.core.impl.testdata.domain.declarative.task_assignment.TestdataTAEmployee;
+import ai.timefold.solver.core.impl.testdata.domain.declarative.task_assignment.TestdataTASchedule;
+import ai.timefold.solver.core.impl.testdata.domain.declarative.task_assignment.TestdataTATask;
+
+import org.junit.jupiter.api.Test;
+
+public class TAShadowVariableTest {
```

Comment:
Let's not overdo it with the tests that run the solver.
I'm fine with having this one, it's a good end-to-end test, but let's test future functionality on a much more granular level.